src/get-started/test-service.md

```diff
@@ -0,0 +1,365 @@
+
+> **Test:** `test_transfer_to_nonexisting_wallet`
+
+Unlike in real Exonum network, you can control which transactions the testkit
+will include into the next block. This allows to test different orderings
+of transactions, even those that would be hard (but not impossible) to reproduce
+in the real network.
+
+Let’s test a case when Alice sends a transaction to Bob while the Bob’s wallet
+is not committed. The test is quite similar to the previous one, with the exception
+how the created transactions are placed into a block.
+Namely, the `create_block_with_transactions` call is replaced with
+
+```rust
+testkit.create_block_with_transactions(txvec![
+    TxCreateWallet::new(&alice_pubkey, "Alice", &alice_key),
+    TxTransfer::new(&alice_pubkey, &bob_pubkey, 10, 0, &alice_key),
+    TxCreateWallet::new(&bob_pubkey, "Bob", &bob_key),
+]);
+```
+
+That is, although Bob's wallet is created, this occurs after the transfer is executed.
+
+We should check that Alice did not send her tokens to nowhere:
+
+```rust
+assert_eq!(alice_wallet.balance(), 100);
+assert_eq!(bob_wallet.balance(), 100);
+```
+
+## Testing API
+
+API-focused tests are placed in a separate file, [`tests/api.rs`][tests-api.rs].
+It is structurally similar to the integration test file we have considered
+previously (including tests), so we will concentrate on differences only.
+
+### API Wrapper
+
+The testkit allows to access service endpoints with the help
+of the [`TestKitApi`][TestKitApi] struct. However, calls to `TestKitApi`
+may be overly verbose and prone to errors for practical purposes,
+as the struct does not know type signatures of the endpoints
+of a specific service. To improve usability
+let’s create a *wrapper* around `TestKitApi` with the wrapper’s methods corresponding
+to service endpoints:
+
+```rust
+struct CryptocurrencyApi {
+    inner: TestKitApi,
+}
+
+impl CryptocurrencyApi {
+    fn create_wallet(&self, name: &str) -> (TxCreateWallet, SecretKey) {
+        // Code skipped...
+    }
+
+    fn transfer(&self, tx: &TxTransfer) {
+        // Code skipped...
+    }
+
+    fn get_wallet(&self, pubkey: &PublicKey) -> Wallet {
+        // Code skipped...
+    }
+}
+```
+
+`create_wallet` returns a secret key along with the created transaction
+because it may be needed to sign other transactions authorized by the wallet owner.
+
+Inside, all wrapper methods call the `inner` API instance; for example, `get_wallet`
+is implemented as
+
+```rust
+fn get_wallet(&self, pubkey: &PublicKey) -> Wallet {
+    self.inner.get(
+        ApiKind::Service("cryptocurrency"),
+        &format!("v1/wallet/{}", pubkey.to_string()),
+    )
+}
+```
+
+That is, the method performs an HTTP GET request with the URL address corresponding
+to a service [with the specified name](../architecture/services.md#service-identifiers)
+and a `v1/wallet/…` path within the service API. When we created the service,
+we [have defined](create-service.md#wire-api) that invoking such a request
```

Comment:
we [have defined](create-service.md#wire-api) that invoking such a request
 +returns information about a specific wallet. --> 
we [defined](create-service.md#wire-api) that invoking such a request
 +would return information about a specific wallet.